src/pivot-table/hooks/use-column-width/use-column-width-left.ts

```diff
@@ -45,47 +73,34 @@ export default function useColumnWidthLeft({ layoutService, tableRect, headersDa
-            const header = row[collIdx];
-            if (!header) return maxWidth;
-
-            const lastRowLastColumn = rowIdx === headersData.size.y - 1 && collIdx === headersData.size.x - 1;
-            let cellWidth = 0;
-
-            if (header.id === PSEUDO_DIMENSION_KEY && header.isLeftDimension) {
-              // Use the max width of all measures
-              cellWidth = maxMeasureCellWidth;
+          let columnWidth: ColumnWidth | undefined;
+          if (lastRowHeader.columnWidthLocation === ColumnWidthLocation.Pivot) {
+            columnWidth = layoutService.layout.qHyperCube.topHeadersColumnWidth;
+          } else if (lastRowHeader.columnWidthLocation === ColumnWidthLocation.Dimension) {
+            ({ columnWidth } = lastRowHeader);
+          }
+
+          let fitToContentWidth = 0;
+          if (lastRowHeader.isLeftDimension) {
+            if (lastRowHeader.id === PSEUDO_DIMENSION_KEY) {
+              fitToContentWidth = measureFitToContentWidth();
             } else {
-              const { label, qApprMaxGlyphCount, columnWidth, isLocked } = header;
-              const expandIconSize =
-                !isFullyExpanded && header.isLeftDimension && collIdx < qNoOfLeftDims - 1 ? EXPAND_ICON_SIZE : 0;
-              const lockedIconSize = isLocked ? LOCK_ICON_SIZE : 0;
-              const measuredHeaderWidth =
-                measureTextForHeader(label) + TOTAL_CELL_PADDING + MENU_ICON_SIZE + lockedIconSize;
-
-              let fitToContentWidth = 0;
-              if (header.isLeftDimension) {
-                fitToContentWidth = Math.max(
-                  measuredHeaderWidth,
-                  estimateWidthForDimensionValue(qApprMaxGlyphCount as number) + TOTAL_CELL_PADDING + expandIconSize,
-                );
-              } else if (lastRowLastColumn && !header.isLeftDimension && hasPseudoDimOnLeft) {
-                fitToContentWidth = maxMeasureCellWidth;
-              } else {
-                fitToContentWidth = measuredHeaderWidth;
-              }
-
-              cellWidth = getColumnWidthValue(tableRect.width, columnWidth, fitToContentWidth);
+              fitToContentWidth = Math.max(
+                dimensionsFitToContentWidth(collIdx),
+                dimensionValuesFitToContentWidth(lastRowHeader, collIdx),
+              );
             }
-
-            // The last cell setting should override the other cells in that column, so we don't pick the max
-            const isTypeAuto = !header.columnWidth || header.columnWidth.type === ColumnWidthType.Auto;
-            if (lastRowLastColumn && !isTypeAuto) {
-              return cellWidth;
+          } else if (!lastRowHeader.isLeftDimension) {
+            fitToContentWidth = dimensionsFitToContentWidth(collIdx);
+            if (lastRowHeader.columnWidthLocation === ColumnWidthLocation.Measures) {
+              fitToContentWidth = measureFitToContentWidth(fitToContentWidth);
```

Comment:
any reason doing not doing this as a if ... else?

Reply:
No reason. I'll change

Reply:
Yes. I need the fitToContent for dimensions before calculating the fitToContent for measures